File: libs/java/core/src/main/java/ca/bc/gov/tno/services/ServiceStatus.java
package ca.bc.gov.tno.services;

import java.util.EnumSet;

import ca.bc.gov.tno.IEnumValue;

/**
 * ServiceStatus enum, provides the service status options.
 */
public enum ServiceStatus implements IEnumValue<String> {
  /**
   * Service is running.
   */
  running("running"),
  /**
   * Service has been requested to pause.
   */
  pause("pause"),
  /**
   * Service is paused.
   */
  paused("paused"),
  /**
   * Service has been requested to resume from pause.
   */
  resume("resume"),
  /**
   * Service is sleeping.
   */
  sleeping("sleeping");

  private final String value;

  /**
   * Creates a new instance of a ServiceStatus enum, initializes with specified
   * parameter.
   * 
   * @param newValue The current value of this enum.
   */
  ServiceStatus(final String newValue) {
    value = newValue;
  }

  /**
   * Get the enum value name.
   */
  public String getValue() {
    return value;
  }

  /**
   * Get all of the service status enum values.
   */
  public static final EnumSet<ServiceStatus> All = EnumSet.allOf(ServiceStatus.class);
}

File: libs/net/services/ServiceStatus.cs
namespace TNO.Services;

/// <summary>
/// ServiceStatus enum, provides status options.
/// </summary>
public enum ServiceStatus
{
    /// <summary>
    /// The service is running.
    /// </summary>
    Running = 0,
    /// <summary>
    /// The service has been asked to pause, but it hasn't paused yet.
    /// </summary>
    RequestPause = 1,
    /// <summary>
    /// The service is paused.
    /// A pause will only occur if a user has requested it.
    /// </summary>
    Paused = 2,
    /// <summary>
    /// The service is has been asked to sleep, but it hasn't gone to sleep yet.
    /// </summary>
    RequestSleep = 3,
    /// <summary>
    /// The service is sleeping.
    /// Sleeping can occur if failures have reached their maximum.
    /// </summary>
    Sleeping = 4,
    /// <summary>
    /// The service has a failure and must stop.
    /// </summary>
    RequestFailed = 5,
    /// <summary>
    /// The service has failed.
    /// </summary>
    Failed = 6,
}
